### PR TITLE
`name` field for snapshots

### PR DESCRIPTION
**Pull request by mike0sv**

### CODE REVIEW

File: src/evidently/suite/base_suite.py [hunk 1]
```diff
@@ -482,11 +483,13 @@ class ReportBase(Display):
     # collection of all possible common options
     options: Options
     id: uuid.UUID
+    name: Optional[str] = None
     timestamp: datetime
     metadata: Dict[str, MetadataValueType] = {}
     tags: List[str] = []
 
-    def __init__(self, options: AnyOptions = None, timestamp: Optional[datetime] = None):
+    def __init__(self, options: AnyOptions = None, timestamp: Optional[datetime] = None, name: str = None):
```
Comment (mou) [inline src/evidently/suite/base_suite.py:491]: The type definition of `name` is different in the model and the argument. But assignment is without any conversion. Is it intended?

Comment (mike0sv) [reply src/evidently/suite/base_suite.py:491]: When you use `None` as default argument value it's ok not to use `Optional` in type annotation

Comment (mou) [reply src/evidently/suite/base_suite.py:491]: I mean why not use `Optional[str]` as type for constructor argument?

Comment (mike0sv) [reply src/evidently/suite/base_suite.py:491]: Because it's shorter? :)

**Pull request merged**